### PR TITLE
add eth_syncing RPC

### PR DESCRIPTION
**Pull request by rakita**
Add `eth_syncing` endpoint: https://eth.wiki/json-rpc/API#eth_syncing

Tested it with infure and return `false` successfully, don't have any syncing node to check return when it is syncing.

### CODE REVIEW

File: ethers-providers/src/lib.rs [hunk 1]
```diff
@@ -79,6 +79,14 @@ where
     }
 }
 
+/// Structure used in eth_syncing RPC
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum SyncingStatus {
+    IsFalse(bool),
```
Comment (gakonst) [inline ethers-providers/src/lib.rs:86]: could you add a note on how this works? not clear to me what `IsFalse(false)` vs `IsFalse(true)` means?

**Review by gakonst — APPROVED** [review]

File: ethers-providers/src/lib.rs (hunk 1 above)
Comment (rakita) [reply ethers-providers/src/lib.rs]: I added comments, but `IsFalse(true)` is not possible to happen by eth spec. wdyt about adding an additional struct, first for deserialization and second for returning to library call?

Comment (gakonst) [reply ethers-providers/src/lib.rs]: oh as in the commit you pushed does not work? yeah I'm OK with adding an intermediate non-public datatype that helps with the desert and then having the "simple" enum.

Comment (rakita) [reply ethers-providers/src/lib.rs]: Yea i made mistake with that commit, now it should work as expected. Can you check it again if it is okay, I embedded intermediate type inside function.

**Review by gakonst — APPROVED** [review]
Gg

**Pull request merged**